drivers/ieee802154/ieee802154_gecko.c

```diff
@@ -0,0 +1,588 @@
+	case IEEE802154_TX_MODE_TXTIME_CCA:
+	default:
+		NET_ERR("TX mode %d not supported", mode);
+		return -ENOTSUP;
+	}
+
+	if (status != RAIL_STATUS_NO_ERROR) {
+		LOG_ERR("Failed to start Tx");
+		return -EIO;
+	}
+
+	/* Wait for the callback from the radio driver. */
+	if (0 != k_sem_take(&dev_data->tx_wait, TX_PACKET_SENT_TIMEOUT)) {
+		LOG_DBG("Failed to take tx_wait semaphore");
+		return -EIO;
+	}
+
+	return dev_data->tx_status;
+}
+
+#define RAIL_IRQ_PRIO  0
+
+void ieee802154_gecko_irq_config(void)
+{
+	IRQ_DIRECT_CONNECT(RFSENSE_IRQn, RAIL_IRQ_PRIO, RFSENSE_IRQHandler, 0);
+	irq_enable(RFSENSE_IRQn);
+	IRQ_DIRECT_CONNECT(AGC_IRQn, RAIL_IRQ_PRIO, AGC_IRQHandler, 0);
+	irq_enable(AGC_IRQn);
+	IRQ_DIRECT_CONNECT(BUFC_IRQn, RAIL_IRQ_PRIO, BUFC_IRQHandler, 0);
+	irq_enable(BUFC_IRQn);
+	IRQ_DIRECT_CONNECT(FRC_IRQn, RAIL_IRQ_PRIO, FRC_IRQHandler, 0);
+	irq_enable(FRC_IRQn);
+	IRQ_DIRECT_CONNECT(FRC_IRQn, RAIL_IRQ_PRIO, FRC_IRQHandler, 0);
+	irq_enable(FRC_IRQn);
+	IRQ_DIRECT_CONNECT(FRC_PRI_IRQn, RAIL_IRQ_PRIO, FRC_PRI_IRQHandler, 0);
+	irq_enable(FRC_PRI_IRQn);
+	IRQ_DIRECT_CONNECT(MODEM_IRQn, RAIL_IRQ_PRIO, MODEM_IRQHandler, 0);
+	irq_enable(MODEM_IRQn);
+	IRQ_DIRECT_CONNECT(PROTIMER_IRQn, RAIL_IRQ_PRIO, PROTIMER_IRQHandler, 0);
+	irq_enable(PROTIMER_IRQn);
+	IRQ_DIRECT_CONNECT(RAC_RSM_IRQn, RAIL_IRQ_PRIO, RAC_RSM_IRQHandler, 0);
+	irq_enable(RAC_RSM_IRQn);
+	IRQ_DIRECT_CONNECT(RAC_SEQ_IRQn, RAIL_IRQ_PRIO, RAC_SEQ_IRQHandler, 0);
+	irq_enable(RAC_SEQ_IRQn);
+	IRQ_DIRECT_CONNECT(SYNTH_IRQn, RAIL_IRQ_PRIO, SYNTH_IRQHandler, 0);
+	irq_enable(SYNTH_IRQn);
+}
+
+static int ieee802154_gecko_init_rail(struct device *dev)
+{
+	struct ieee802154_gecko_context *dev_data = DEV_DATA(dev);
+	const struct ieee802154_gecko_dev_cfg *const dev_cfg = DEV_CFG(dev);
+	RAIL_Status_t status;
+
+	dev_data->rail_handle = RAIL_Init(&rail_config, NULL);
+	if (dev_data->rail_handle == NULL) {
+		LOG_DBG("Failed to get RAIL handle");
+		return -EIO;
+	}
+	RAIL_Idle(dev_data->rail_handle, RAIL_IDLE, true);
+
+	RAIL_EnablePaCal(true);
+	RAIL_ConfigCal(dev_data->rail_handle, RAIL_CAL_ALL);
+
+	/* Configure RAIL callbacks */
+	RAIL_ConfigEvents(dev_data->rail_handle, RAIL_EVENTS_ALL,
+			  RAIL_EVENT_RX_PACKET_RECEIVED |
+			  RAIL_EVENT_TX_PACKET_SENT |
+			  RAIL_EVENT_TX_ABORTED |
+			  RAIL_EVENT_TX_BLOCKED |
+			  RAIL_EVENT_TX_UNDERFLOW |
+			  RAIL_EVENT_TX_CHANNEL_BUSY |
+			  RAIL_EVENT_CAL_NEEDED);
+
+	/* Initialize the PA */
+	RAIL_TxPowerConfig_t txPowerConfig = {
+		.mode = RAIL_TX_POWER_MODE_2P4_HP,
+		.voltage = 1800U,
+		.rampTime = 10U,
+	};
+	status = RAIL_ConfigTxPower(dev_data->rail_handle, &txPowerConfig);
+	if (status != RAIL_STATUS_NO_ERROR) {
+		LOG_DBG("RAIL_ConfigTxPower returned an error %d", status);
+		return -EIO;
+	}
+
+	status = RAIL_SetTxPower(dev_data->rail_handle,
+				 dev_cfg->init_tx_power_level_raw);
+	if (status != RAIL_STATUS_NO_ERROR) {
+		LOG_DBG("RAIL_SetTxPower returned an error %d", status);
+		return -EIO;
+	}
+
```

Comment:
This might be needed, although I not sure.
```suggestion
const RAIL_DataConfig_t rail_data_cfg = {
	.txSource = TX_PACKET_DATA,
	.rxSource = RX_PACKET_DATA,
	.txMethod = PACKET_MODE,
	.rxMethod = PACKET_MODE,
};
status = RAIL_ConfigData(gecko_radio->rail_handle, &rail_data_cfg);
if (status != RAIL_STATUS_NO_ERROR) {
	LOG_DBG("RAIL_ConfigData returned an error %d", status);
	return -EIO;
}
```